clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -149,92 +212,84 @@ private boolean maybeTransitionToStable() {
         return state.equals(MemberState.STABLE);
     }
 
+    /**
+     * Take new target assignment received from the server and set it as targetAssignment to be
+     * processed. If an assignment is already in process this newTargetAssignment will be ignored
+     * for now.
+     */
     private void setTargetAssignment(ConsumerGroupHeartbeatResponseData.Assignment newTargetAssignment) {
         if (!targetAssignment.isPresent()) {
             targetAssignment = Optional.of(newTargetAssignment);
         } else {
-            // Keep the latest next target assignment
-            nextTargetAssignment = Optional.of(newTargetAssignment);
+            log.debug(String.format("Temporarily ignoring assignment %s received while member %s " +
```

Comment:
We don't need the `String.format` here, do we? We should be able to rely on the placeholder syntax in the `log.debug` directly.

Reply:
Totally!